### PR TITLE
HBSP 335 support custom widgets in blog posts

### PR DESCRIPTION
**Pull request by kdudina-smartling**
MERGED from HBSP-326-Blog-post-processing

### CODE REVIEW

File: src/main/java/com/smartling/connector/hubspot/sdk/rest/api/BlogPostsRawApi.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package com.smartling.connector.hubspot.sdk.rest.api;
+
+import feign.Headers;
+import feign.Param;
+import feign.RequestLine;
+
+public interface BlogPostsRawApi
+{
+    @RequestLine("GET /content/api/v2/blog-posts/{blog_post_id}")
```
Comment (dimitrystd) [inline src/main/java/com/smartling/connector/hubspot/sdk/rest/api/BlogPostsRawApi.java:9]: Why v2 when hubspot uses v3. Please check that new version works by tests.
We implement a brand new functionality that will be fully tested manually. So why lose the opportunity and use legacy API that someone will have to update (when hubspot will send a notice)
image

**Review by dimitrystd — CHANGES_REQUESTED** [review]
Left comments in another review. I expect that some methods will be deleted as unused

File: src/main/java/com/smartling/connector/hubspot/sdk/rest/api/BlogPostsRawApi.java (hunk 1 above)
Comment (sl-mmuradov) [reply src/main/java/com/smartling/connector/hubspot/sdk/rest/api/BlogPostsRawApi.java:9]: We use V2 API for now. I would migrate to V3 API by separate task/PR.

Comment (kdudina-smartling) [reply src/main/java/com/smartling/connector/hubspot/sdk/rest/api/BlogPostsRawApi.java:9]: V3 is in beta currently - https://developers.hubspot.com/docs/api/cms/blog-post?_ga=2.114834759.1334556895.1610638711-771452351.1610450942. Also model changed a lot.

Comment (dimitrystd) [reply src/main/java/com/smartling/connector/hubspot/sdk/rest/api/BlogPostsRawApi.java:9]: ok, this is a solid reason. Someone will be unlucky in the future :)

File: src/integrationTest/java/com/smartling/connector/hubspot/sdk/it/BlogPostsIntegrationTest.java [hunk 2]
```diff
@@ -147,50 +170,30 @@ public void shouldThrowExceptionIfAuthorizationFailed() throws HubspotApiExcepti
...
         BlogPostDetail blogPostById = hubspotClient.getBlogPostById(BLOG_POST_ID2);
 
         blogPostById.setId(otherBlogPostId);
+        String blogPostJson = OBJECT_MAPPER.writeValueAsString(blogPostById);
 
-        hubspotClient.updateBlogPost(blogPostById);
-    }
-
-    @Test
-    public void shouldCloneBlogPost() throws HubspotApiException {
```
Comment (dimitrystd) [inline src/integrationTest/java/com/smartling/connector/hubspot/sdk/it/BlogPostsIntegrationTest.java:173]: We still have Clone and Delete methods in API but no simple ways to run tests for these methods anymore.

File: src/main/java/com/smartling/connector/hubspot/sdk/rest/api/BlogPostsRawApi.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    String blogPost(@Param("blog_post_id") String blogPostId);
+
+    @RequestLine("POST /content/api/v2/blog-posts")
+    @Headers("Content-Type: application/json")
+    String create(String blogPostAsJson);
```
Comment (dimitrystd) [inline src/main/java/com/smartling/connector/hubspot/sdk/rest/api/BlogPostsRawApi.java:14]: Nitpick. The new raw api uses another method name pattern (inconsistent with another API client). Compare new `create` with old `createBlogPost`.

**Review by dimitrystd — CHANGES_REQUESTED** [review]

**Review by sl-mmuradov — APPROVED** [review]

File: src/main/java/com/smartling/connector/hubspot/sdk/rest/api/BlogPostsRawApi.java (hunk 3 above)
Comment (kdudina-smartling) [reply src/main/java/com/smartling/connector/hubspot/sdk/rest/api/BlogPostsRawApi.java]: Removed create method at all because it is not needed anymore

**Review by dkBrazz — APPROVED** [review]

**Pull request merged**